sinker/config_state_check.go

```diff
@@ -0,0 +1,57 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+package sinker
+
+import (
+	"github.com/ns1labs/orb/sinker/config"
+	"go.uber.org/zap"
+	"time"
+)
+
+const (
+	streamID       = "orb.sinker"
+	streamLen      = 1000
+	CheckerFreq    = 300 * time.Second
+	DefaultTimeout = 30 * time.Minute
+)
+
+func (svc *sinkerService) checkState(t time.Time) {
+	owners, err := svc.sinkerCache.GetAllOwners()
+	if err != nil {
+		svc.logger.Error("failed to retrieve the list of owners")
+		return
+	}
+
+	for _, ownerID := range owners {
+		configs, err := svc.sinkerCache.GetAll(ownerID)
+		if err != nil {
+			svc.logger.Error("unable to retrieve policy state", zap.Error(err))
+			return
+		}
+		for _, cfg := range configs {
+			// Set idle if the sinker is more then 30 minutes not sending metrics (Remove from Redis)
+			if cfg.LastRemoteWrite.Add(DefaultTimeout).Before(time.Now()) {
+				if cfg.State == config.Active {
+					if err := svc.sinkerCache.Remove(cfg.OwnerID, cfg.SinkID); err != nil {
```

Comment:
did you confirm this won't fail if it's already been removed? we know that the different sinker nodes will race to do this, and only one will succeed

Reply:
If Redis try to remove a non-existing key, it's ignored.